### PR TITLE
[RFC] configure script + Logind providers

### LINKED ISSUE #67: [PATCH] Add elogind support
```patch
From e00637efaeb5c8b429b40fd45dd4598544df871a Mon Sep 17 00:00:00 2001
From: Andrew Tropin <andrew@trop.in>
Date: Tue, 25 May 2021 11:08:10 +0300
Subject: [PATCH] Add elogind support

<systemd/sd-bus.h> can be provided not only by libsystemd, but also by
libelogind, so lookup for libelogind too.

---
Probably it's also a reasonable idea to rename ENABLE_SYSTEMD to
ENABLE_LOGIND

 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 105f5d6..82b8595 100644
--- a/Makefile
+++ b/Makefile
@@ -18,6 +18,8 @@ MODE = ${MODE_${INSTALL_UDEV_RULES}}
 ifdef ENABLE_SYSTEMD
 	CFLAGS += ${shell pkg-config --cflags libsystemd}
 	LDLIBS += ${shell pkg-config --libs libsystemd}
+	CFLAGS += ${shell pkg-config --cflags libelogind}
+	LDLIBS += ${shell pkg-config --libs libelogind}
 	CPPFLAGS += -DENABLE_SYSTEMD
 	INSTALL_UDEV_RULES=0
 	MODE = 0755
-- 
2.31.1
```

### PR DESCRIPTION
**Pull request by xdbob**
As I suggested in #67 I converted the build-system to the traditional `./configure && make && sudo make install` (without the whole autotools nonsense).

What do you think ?

And btw, neither `basu` nor `elogind` are packaged on my distribution someone need to test if they work (and are configured) as expected.

Closes #67 

PS: @abcdw I credited you in `dbus: support elogind and basu at build time` if you want a more explicit reference I can credit you in the code or not at all if you wish

### CODE REVIEW

**Comment by xdbob** [comment]
I would also like to change 2 of the defaults:
* enable logind -> most of the distributions should have systemd v243 by now
* prefix=/usr/local -> let's keep `/usr` for distribution packages

Are you OK with that @Hummer12007 ?

File: configure [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+# Specialize automatic options
+if [ "$udev" = "auto" ]; then
+	if [ "$logind" = "disabled" ]; then
+		udev="enabled"
+	else
+		udev="disabled"
+	fi
+fi
+
+if [ "x$mode" = "xauto" ]; then
```
Comment (Hummer12007) [inline configure:78]: Let's get rid of `x` here

**Comment by Hummer12007** [comment]
Thanks! Mind pinging me if I don't get to the PR in the next few days?

**Comment by abcdw** [comment]
It will require some changes to package recipe in downstream package managers and distros, but overall looks good to me and such approach seems a little cleaner.

**Comment by xdbob** [comment]
> Thanks! Mind pinging me if I don't get to the PR in the next few days?

@Hummer12007 ping ? :)

btw, you did not address my other question:

> I would also like to change 2 of the defaults:
> 
>     * enable logind -> most of the distributions should have systemd v243 by now
> 
>     * prefix=/usr/local -> let's keep `/usr` for distribution packages
> 
> 
> Are you OK with that @Hummer12007 ?

**Comment by Hummer12007** [comment]
@xdbob looks good?

**Comment by xdbob** [comment]
> @xdbob looks good?

I would reverse the bus provider checks and break when the first one is found, eg:

```sh
for prov in libsystemd libelogind basu; do
    if pkg-config --exists "$prov" >/dev/null 2>&1; then
        bus_provider="$prov"
        sdbus="$prov"
        break
    fi
done
```

Otherwise, LGTM :+1:

**Comment by Hummer12007** [comment]
Nice catch! The order was intended.
With regards to enabling logind by default, I'd rather not (it is much slower than the ordinary sysfs setter, which won't play nice with the gradual brightness transition). It is up to the distros to decide anyway.
With regards to the default prefix, `/usr/local` is the better option, as the packagers *should* explicitely specify the prefix anyway.

**Comment by Hummer12007** [comment]
Oh, you suggested /usr/local as the new default.

**Comment by Hummer12007** [comment]
Thanks!

**Pull request merged**